python/seldon_core/metrics.py

```diff
@@ -1,10 +1,115 @@
+from prometheus_client.core import (
+    HistogramMetricFamily,
+    GaugeMetricFamily,
+    CounterMetricFamily,
+    CollectorRegistry,
+)
+from prometheus_client import exposition
+from prometheus_client.utils import floatToGoString
+
+from multiprocessing import Manager
+
+import numpy as np
+
 from typing import List, Dict
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
 
 COUNTER = "COUNTER"
 GAUGE = "GAUGE"
 TIMER = "TIMER"
 
 
+def generate_metrics(metrics):
+    myregistry = CollectorRegistry()
+    myregistry.register(metrics)
+    return (
+        exposition.generate_latest(myregistry).decode("utf-8"),
+        exposition.CONTENT_TYPE_LATEST,
+    )
+
+
+BINS = [0] + list(np.logspace(-3, np.log10(30), 50)) + [np.inf]
+LABELS = ["worker-id", "model", "image"]
+
+my_labels = {"model": "latest", "image": "my-image"}
+
+
+def update_hist(x, vals, sumv):
+    hist = np.histogram([x], BINS)[0]
+    vals = list(np.array(vals) + hist)
+    return vals, sumv + x
+
+
+def expose_gauge(name, value, labels):
+    metric = GaugeMetricFamily(name, "", labels=LABELS)
+    metric.add_metric(labels, value)
+    return metric
+
+
+def expose_counter(name, value, labels):
+    metric = CounterMetricFamily(name, "", labels=LABELS)
+    metric.add_metric(labels, value)
+    return metric
+
+
+def expose_histogram(name, value, labels):
```

Comment:
If these functions are not used outside of the file and only in the class can we make them either static or normal class functions?

Reply:
Same for the ones above